src/Microsoft.Health.Fhir.SqlServer/Features/Watchdogs/CleanupEventLogWatchdog.cs

```diff
@@ -57,5 +69,204 @@ INSERT INTO dbo.Parameters (Id,Char) SELECT 'CleanpEventLog', 'LogEvent'
+  FROM sys.indexes I
+  WHERE EXISTS (SELECT * FROM sys.partition_schemes PS WHERE PS.data_space_id = I.data_space_id AND PS.name = 'PartitionScheme_ResourceTypeId')
+    AND object_id <> object_id('Resource')
+    AND index_id = 1
+            ");
+            return await sqlCommand.ExecuteReaderAsync(_sqlRetryService, reader => reader.GetString(0), _logger, cancellationToken);
+        }
+
+        private async Task LogRawResourceStats(CancellationToken cancellationToken)
+        {
+            try
+            {
+                var st = DateTime.UtcNow;
+
+                await CreateTmpProceduresAsync(cancellationToken);
+
+                var maxSurrogateId = await GetMaxSurrogateId(cancellationToken);
+                _logger.LogInformation($"DatabaseStats.MaxSurrogateId={maxSurrogateId}");
+
+                var types = await GetUsedResourceTypesAsync(cancellationToken);
+                _logger.LogInformation($"DatabaseStats.UsedResourceTypes.Count={types.Count}");
+
+                foreach (var type in types)
+                {
+                    var totalCompressedBytes = 0L;
+                    var totalDecompressedBytes = 0L;
+                    var totalResources = 0L;
+                    var surrogateId = 0L;
+                    IReadOnlyList<(long SurrogateId, byte[] RawResourceBytes)> rawResources;
+                    do
+                    {
+                        rawResources = await GetRawResourcesAsync(type, surrogateId, maxSurrogateId, cancellationToken);
+                        foreach (var rawResource in rawResources)
+                        {
+                            surrogateId = rawResource.SurrogateId;
+                            var rawResourceBytes = rawResource.RawResourceBytes;
+                            var isInvisible = rawResourceBytes.Length == 1 && rawResourceBytes[0] == 0xF;
+                            if (!isInvisible)
+                            {
+                                totalCompressedBytes += rawResourceBytes.Length;
+                                using var rawResourceStream = new MemoryStream(rawResourceBytes);
+                                totalDecompressedBytes += UTF8Encoding.UTF8.GetBytes(_compressedRawResourceConverter.ReadCompressedRawResource(rawResourceStream)).Length;
+                                totalResources++;
+                            }
+                        }
+                    }
+                    while (rawResources.Count > 0);
+
+                    var totals = new ResourceTypeTotals { ResourceTypeId = type, Resources = totalResources, CompressedBytes = totalCompressedBytes, DecompressedBytes = totalDecompressedBytes };
+                    var totalsStr = JsonSerializer.Serialize(totals);
+                    _logger.LogInformation($"DatabaseStats.ResourceTypeTotals={totalsStr}");
+                    await _sqlRetryService.TryLogEvent("DatabaseStats.ResourceTypeTotals", "Warn", totalsStr, st, cancellationToken);
```

Comment:
Why log these as warnings to the event log?

Reply:
LogEvent call requires special record in the Parameters table for any status other than warn on error.